### PR TITLE
Adding last fixes and corrections to the StructuralMechanicsApplication in the release

### PR DESCRIPTION
**Pull request by loumalouomega**

### CODE REVIEW

**Comment by loumalouomega** [comment]
The last geometries are included because it is needed for computing the Normal in the new load implementations

**Comment by loumalouomega** [comment]
@RiccardoRossi I added just the geometries that include the method Normal(). geometry.h, triangle3d_3 and line2d_2

File: kratos/geometries/geometry.h [hunk 1]
```diff
@@ -667,21 +667,67 @@ class Geometry : public PointerVector<TPointType>
...
+        else
+        {
+            for (unsigned int i_dim = 0; i_dim < dimension; i_dim++)
+            {
+                tangent_xi[i_dim]  = j_node(i_dim, 0);
+                tangent_eta[i_dim] = j_node(i_dim, 1);
+            } 
+        }
 
+        return MathUtils<double>::UnitCrossProduct( tangent_eta, tangent_xi );
```
Comment (RiccardoRossi) [inline kratos/geometries/geometry.h:728]: UnitCrossProduct should not be used...

File: kratos/geometries/line_2d_2.h [hunk 2]
```diff
@@ -830,85 +855,67 @@ class Line2D2 : public Geometry<TPointType>
...
-//        KRATOS_WATCH(dist_proy);
-
-        if (dist_proy < tol)
-        {
-            const double L  = Length();
+        const double Tolerance = 1e-14; // Tolerance
 
-            const double l1 = std::sqrt((rPoint[0] - FirstPoint[0]) * (rPoint[0] - FirstPoint[0])
-                      + (rPoint[1] - FirstPoint[1]) * (rPoint[1] - FirstPoint[1]));
+        const double Length = this->Length();
```
Comment (RiccardoRossi) [inline kratos/geometries/line_2d_2.h:900]: this is a virtual call. The previousverson was faster (change this in the master, not in the release)

**Review by RiccardoRossi — COMMENTED** [review]
I tried the release and it works for me (i compiled and run the structural mechanics tests in windows. BTW, there are still troubles with running all the tests in win. i'll take a look at it tomorrow. ). I would give a +1. However this would imply compiling everything once again and recheckign the applications. So i think it is @roigcarlo that should decide if he is willing to do it once again. For this reason i'll wait to approve.

**Review by RiccardoRossi — COMMENTED** [review]
I tried the release and it works for me (i compiled and run the structural mechanics tests in windows. BTW, there are still troubles with running all the tests in win. i'll take a look at it tomorrow. ). I would give a +1. However this would imply compiling everything once again and recheckign the applications. So i think it is @roigcarlo that should decide if he is willing to do it once again. For this reason i'll wait to approve.

**Review by roigcarlo — APPROVED** [review]
No problem for me to compile again. If the application works or not, that's beyond my understanding so I'll trust you on that :+1: (if you give me a case I can run it  in a clean machine)

File: kratos/geometries/geometry.h (hunk 1 above)
Comment (loumalouomega) [reply kratos/geometries/geometry.h:728]: I will remove it

File: kratos/geometries/line_2d_2.h (hunk 2 above)
Comment (loumalouomega) [reply kratos/geometries/line_2d_2.h:900]: Can I change it in both?

Comment (loumalouomega) [reply kratos/geometries/line_2d_2.h:900]: I have done in #682, do I cherrypick?

**Pull request merged**